### ISSUE TITLE
[Question] Is it normal for the fileTypes facet to be stored in fixedFacets ?

### ISSUE THREAD

**Issue by JonathanMontane** [raml-org/raml-js-parser-2#597]
**version: 1.1.12**

the following data-type
```yaml
  Photo:
    type: file
    fileTypes: [ 'image/jpeg', 'image/png' ]
    maxLength: 1048576
```

is transformed into

```js
    {
      "Photo": {
        "name": "Photo",
        "displayName": "Photo",
        "typePropertyKind": "TYPE_EXPRESSION",
        "type": [
          "file"
        ],
        "fileTypes": [
          {},
          {}
        ],
        "maxLength": 1048576,
        "fixedFacets": {
          "fileTypes": {
            "enum": [
              "image/jpeg",
              "image/png"
            ]
          },
          "maxLength": 1048576
        },
        "__METADATA__": {
          "primitiveValuesMeta": {
            "displayName": {
              "calculated": true
            }
          }
        }
      }
    }
```

Why are the `fileTypes` and `maxLength` values stored in `fixedFacets` ? (also, why is `maxLength` duplicated between the `Photo.maxLength` object and the `fixedFacets.maxLength` field)

Why is `Photo.fileTypes` equal to `[ {}, {} ]` ?

Thanks.

**Comment by ddenisenko** [comment]
>Why are the fileTypes and maxLength values stored in fixedFacets
Because `fixed` facet is a facet defined in a supertype and being `fixed` (assigned a value) in a descendant type.

>(also, why is maxLength duplicated between the Photo.maxLength object and the fixedFacets.maxLength field)

Yeah, looks being redundant. @KonstantinSviridov @sichvoge what do you guys think of it? Do we actually need to print a list of fixed facets separately?

>Why is Photo.fileTypes equal to [ {}, {} ] ?
Looks like a bug for me.

**Comment by JonathanMontane** [comment]
@ddenisenko Thanks for the clarification! This helped a lot.

**Comment by KonstantinSviridov** [comment]
Yep, it's a bug. In this case `fixedFacets` property should be undefined as it is intended for user defined facets only, while `fileTypes` and `maxLength` are built-in ones and, thus, are represented by separate properties.

**Comment by KonstantinSviridov** [comment]
Hi, @JonathanMontane 

The issue will be fixed in 1.1.14. Now the fix is available in the `develop` branches set.

Regards, Konstantin

**Comment by JonathanMontane** [comment]
@KonstantinSviridov thanks!
